### PR TITLE
Hot reload

### PR DESCRIPTION
**Pull request by christianalfoni**
This allows you to run the client with:

```sh
yarn start:hot
```

It:

- Runs webpack in hot mode
- Runs `next` version of Overmind which has fixed hot reloading (still want to make a fix for actions, though not criticial)
- It does not load the sandbox and embed, as those heavily affect the rebuild time on changes

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/4111/commits/4845146fb2eb8621bf1d2c0f6e3d3900e401b69a is at https://pr4111.build.csb.dev/s/new.

File: packages/app/config/webpack.common.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const babelProd = require('./babel.prod');
 
 const NODE_ENV = JSON.parse(env.default['process.env.NODE_ENV']);
 const SANDBOX_ONLY = !!process.env.SANDBOX_ONLY;
+const APP_HOT = !!process.env.APP_HOT;
```
Comment (MichaelDeBoey) [inline packages/app/config/webpack.common.js:21]:
```suggestion
const APP_HOT = Boolean(process.env.APP_HOT);
```

File: packages/app/config/webpack.dev.js [hunk 2]
```diff
@@ -4,18 +4,20 @@ const merge = require('webpack-merge');
...
 const commonConfig = require('./webpack.common');
 
-const devEntries = [
-  // 'react-hot-loader/patch',
-  'webpack-dev-server/client?/',
-  // 'webpack/hot/only-dev-server',
-];
+const devEntries = ['webpack-dev-server/client?/'];
+
+const APP_HOT = !!process.env.APP_HOT;
```
Comment (MichaelDeBoey) [inline packages/app/config/webpack.dev.js:9]:
```suggestion
const APP_HOT = Boolean(process.env.APP_HOT);
```

**Comment by CompuIves** [comment]
Nice!!! Is this Fast Refresh?

**Comment by christianalfoni** [comment]
Nopes, not `fast-refresh`, just normal as `fast-refrech` is still experimental

**Pull request merged**